### PR TITLE
[go/conformance] Fix l2-resource-config conformance test

### PR DESCRIPTION
**Pull request by Zaid-Ajaj**
### Description

This PR fixes l2-resource-config conformance test in Go. The problem was that we weren't emitting the version of the package as part the default resource options unless we have `respectSchemaVersion: true` in the Go language options in our test schemas. The fix is to emit the exact version not only when  `respectSchemaVersion: true` but also when the package supports packing and the version is specified (as the case with the conformance test providers)

### CODE REVIEW

File: pkg/codegen/go/gen.go [hunk 1]
```diff
@@ -5246,6 +5246,8 @@ func Pkg%[1]sDefaultOpts(opts []pulumi.%[1]sOption) []pulumi.%[1]sOption {
 		if info.(GoPackageInfo).RespectSchemaVersion && pkg.pkg.Version() != nil {
 			versionPackageRef = fmt.Sprintf("semver.MustParse(%q)", p.Version.String())
 		}
+	} else if pkg.pkg.SupportPack() && pkg.pkg.Version() != nil {
+		versionPackageRef = fmt.Sprintf("semver.MustParse(%q)", p.Version.String())
 	}
```
Comment (Zaid-Ajaj) [inline pkg/codegen/go/gen.go:5251]: The actual fix

File: sdk/go/pulumi-language-go/language_test.go [hunk 2]
```diff
@@ -170,10 +170,8 @@ func runTestingHost(t *testing.T) (string, testingrpc.LanguageTestClient) {
 	return engineAddress, client
 }
 
-// TODO: These tests are not working yet because of issues in sdkgen and programgen
-var expectedFailures = map[string]string{
-	"l2-resource-config": "expected no error, got BAIL: unexpected provider request with no version",
-}
+// Add test names here that are expected to fail and the reason why they are failing
+var expectedFailures = map[string]string{}
```
Comment (Zaid-Ajaj) [inline sdk/go/pulumi-language-go/language_test.go:174]: No more failing tests 🎉

File: pkg/codegen/go/gen.go (hunk 1 above)
Comment (Frassle) [reply pkg/codegen/go/gen.go:5251]: If SupportPack is true then we should error if Version is nil

Comment (Zaid-Ajaj) [reply pkg/codegen/go/gen.go:5251]: That should probably happen during schema bind though, not sdk-gen, I can open another PR for that

Comment (Frassle) [reply pkg/codegen/go/gen.go:5251]: It probably should, I think I've done err checks in all the other sdk gens though. Happy for you to clean them up and defer to bind doing the right thing.

Comment (Zaid-Ajaj) [reply pkg/codegen/go/gen.go:5251]: I haven't looked at all validations yet but for this one here https://github.com/pulumi/pulumi/pull/17420

**Review by Frassle — APPROVED** [review]

**Pull request merged**